### ISSUE TITLE
Swift Compilation Issue when property named 'fields'

### ISSUE THREAD

**Issue by ethansinjin** [apollographql/apollo-ios#103]
My project has a query `userDetailGroups(first: Int! = 10, after: String = null): UserDetailGroupsConnection`.

Here is the definition for a UserDetailGroup:
```
type UserDetailGroup implements Node {
  fields(first: Int = 10, after: String = null): UserDetailFieldsConnection
}
```

In my Xcode project, I'm attempting the following:

```
query UserDetailGroups($userDetailGroupsFirst: Int = 10, $userDetailGroupsAfter: String = null, $fieldsFirst: Int = 10, $fieldsAfter: String = null) {
    userDetailGroups(first: $userDetailGroupsFirst, after: $userDetailGroupsAfter) {
        pageInfo {
            hasPreviousPage
            hasNextPage
        }
        edges {
            cursor
            node {
                id
                fields(first: $fieldsFirst, after: $fieldsAfter) {
                    pageInfo { ... }
                    edges { ... }
                }
            }
        }
    }
}
```

This results in the following error from the Swift compiler:

"Expression type UserDetailGroupsQuery.Data.UserDetailGroup.Edge.Node.Field is ambiguous without more context"

Xcode version:  8.3.3 (8E3004b)
macOS version: 10.12.5 (16F73)
node version: v7.4.0 (npm v5.0.4)
iOS 10.3.1 SDK
Using Cocoapods for Apollo. (pod 'Apollo', '~> 0.6.0-beta.1')
apollo-codegen 0.13.2

(On a side note, it looks like 0.6.0 has a tag on this repository but can't be downloaded using CocoaPods without the beta suffix.)

**Comment by ethansinjin** [comment]
I have confirmed that the name 'fields' is specifically what triggers this issue. If I rename the query on the server to something else, the errors disappear.

**Comment by martijnwalraven** [comment]
This should be fixed in Apollo iOS 0.6.2.
